### PR TITLE
Added new tutorial for `ESP32 circuit`

### PR DESCRIPTION
**Pull request by AnasSarkiz**
/claim #39

### CODE REVIEW

**Comment by AnasSarkiz** [comment]
https://github.com/user-attachments/assets/7f00a9fd-ae46-4776-9dd7-39235a67e9b3

File: tutorials/esp32-circuit.mdx [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+
+```tsx
+import { useESP32 } from "@tsci/AnasSarkiz.ESP32"
+import { useResistor, useCapacitor } from "@tscircuit/core"
+
+export const Circuit = () => {
+  const ESP32 = useESP32("U2")
+
+  // Capacitors
+  const C1 = useCapacitor("C1", { capacitance: "0F" }) //TDB
```
Comment (seveibar) [inline tutorials/esp32-circuit.mdx]: Are these really 0F?

File: tutorials/esp32-circuit.mdx [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+/>
+```
+
+## Memory Components
+
+The circuit includes both Flash and PSRAM memory:
+
+```tsx
+<chip
+  name="U3" // Flash Memory
```
Comment (seveibar) [inline tutorials/esp32-circuit.mdx]: I wonder if we could get a manufacturer part number for this

**Review by seveibar — APPROVED** [review]
Note comments, need to check schematic against reference still but this is a well sized incremental step

File: tutorials/esp32-circuit.mdx (hunk 1 above)
Comment (AnasSarkiz) [reply tutorials/esp32-circuit.mdx]: Screenshot 2024-12-04 at 10 32 23 AM

File: tutorials/esp32-circuit.mdx (hunk 2 above)
Comment (AnasSarkiz) [reply tutorials/esp32-circuit.mdx]: we already have
Screenshot 2024-12-04 at 10 36 55 AM

File: tutorials/esp32-circuit.mdx (hunk 1 above)
Comment (seveibar) [reply tutorials/esp32-circuit.mdx]: oh i missed this...

File: tutorials/esp32-circuit.mdx [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+---
+
+## Overview
+
+In this tutorial, we'll create a development circuit based on the ESP32 microcontroller. This circuit includes all essential components needed for the ESP32 to function properly, including decoupling capacitors, crystal oscillator, flash memory, and antenna components.
+
+## Components Overview
+
+The main components in this circuit are:
+- ESP32 microcontroller
```
Comment (seveibar) [inline tutorials/esp32-circuit.mdx]: ESP32 is actually a family of microcontrollers, can you name the specific manufacturer part number? Goes for all the components here- they should have specific manufacturer part numbers

File: tutorials/esp32-circuit.mdx [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+```tsx
+import { ESP32Module } from "../components/ESP32Module"
+```
+
+## Basic Circuit Structure
+
+The ESP32 module is implemented as a reusable component that includes all the necessary supporting circuitry. Let's examine how to use it:
+
+```tsx
+import { useESP32 } from "@tsci/AnasSarkiz.ESP32"
```
Comment (seveibar) [inline tutorials/esp32-circuit.mdx]: this module should be named more specific to the actual ESP32 part number

File: tutorials/esp32-circuit.mdx (hunk 1 above)
Comment (seveibar) [reply tutorials/esp32-circuit.mdx]: yea it depends on the antenna, 0F for now is probably ok

**Review by seveibar — CHANGES_REQUESTED** [review]
let's add specific part numbers w/ the `manufacturerPartNumber` prop

**Review by seveibar — APPROVED** [review]

**Pull request merged**